src/mpi/romio/adio/ad_daos/ad_daos_io_str.c

```diff
@@ -70,7 +70,8 @@ ADIOI_DAOS_StridedListIO(ADIO_File fd, const void *buf, MPI_Aint count,
                          MPI_Request * request, int rw_type, int *error_code)
 {
     ADIOI_Flatlist_node *flat_buf, *flat_file;
-    int i, j, k, fwr_size = 0, st_index = 0;
+    int fwr_size = 0, st_index = 0;
+    size_t i, j, k;
```

Comment:
Let's remove declaration of these and adopt `for (TYPE i = ...)`. There is no way we can understand why these variables need be certain type until we see its usage.